integration-tests/kotlin/src/test/resources/projects/kotlin-compiler-args/pom.xml

```diff
@@ -62,9 +66,9 @@
                 <artifactId>quarkus-maven-plugin</artifactId>
                 <version>${quarkus.version}</version>
                 <configuration>
-                    <compilerArgs>
-                        <arg>-Xallow-result-return-type</arg>
-                    </compilerArgs>
+                    <kotlinCompilerArgs>
+                        <arg>-Xemit-jvm-type-annotations</arg>
+                    </kotlinCompilerArgs>
```

Comment:
Now that we have a generic Java API for provider-specific compiler args, our config doesn't look flexible enough. This could be a separate issue. But we need to be careful if we want to refactor the config in the near future, to not end up supporting two ways of providing the same configuration. What I mean is we could have something like:
````
<configuration>
    <compiler>
        <kind>java</kind>
        <arg>xxx</arg>
        <arg>yyy</arg>
    </compiler>
    <compiler>
        <kind>kotlin</kind>
        <arg>xxx</arg>
        <arg>yyy</arg>
    </compiler>
</configuration>
````

Reply:
I think we can merge this PR and follow up. I could help with that assuming we agree on how we want the config to look like. The thing is that we also have scala compiler provider, for example. And if somebody adds another one, will we keep adding `scalaCompilerArgs`, `otherCompilerArgs`, etc?

Reply:
What kind of breaking changes would it introduce? We could still support the current `compilerArgs` by interpreting it as the Java compiler ones and logging a warning asking to adjust to the new way of configuring them.

Reply:
For the gradle plugin we could have something like (we may want to extract compiler name as constant somehow):

    quarkusDev {
        compilersOptions {
            compiler("java").options(["-parameters"])
            compiler("kotlin").option("--Xemit-jvm-type-annotations")
        }
    }

For maven we could use the same syntax as manifest, with

    <configuration>
        <compilersOptions>
           <compiler>
              <name>java</name>
              <arg>-parameters</arg>
           </compiler>
           <compiler>
              <name>kotlin</name>
              <arg>--Xemit-jvm-type-annotations</arg>
           </compiler>
        </compilersOptions>
    </configuration> 

WDYT ?